pgjdbc/src/test/java/org/postgresql/util/ReaderInputStreamTest.java

```diff
@@ -182,6 +184,26 @@ public void malformedSingleCharArrayReadTest() throws IOException {
     read(is, 0xF0, 0xA0, 0x9C, 0x8E);
   }
 
+  @Test
+  public void readsSmallerThanBlockSizeTest() throws Exception {
+    final int BLOCK = 8 * 1024;
+    final int DATASIZE = BLOCK * 5 + 57;
+    final byte[] data = new byte[DATASIZE];
+    final byte[] buffer = new byte[BLOCK];
+
+    InputStreamReader isr = new InputStreamReader(new ByteArrayInputStream(data));
+    ReaderInputStream r = new ReaderInputStream(isr);
```

Comment:
`DEFAULT_CHAR_BUFFER_SIZE` should be private, and it should not be accessed/visible from external point of view.